lib/deliveries_manager/lib/deliveries_manager.rb

```diff
@@ -3,5 +3,24 @@
 require 'deliveries_manager/engine'
 
 module DeliveriesManager
-  # Your code goes here...
+  def self.processPayload(payload)
+    orderPayload = JSON.parse(payload)
+
+    if (!OrderProcessing.exists?(:order_id => orderPayload["order_id"]))
+      savedOP = OrderProcessing.new(orderPayload)
+      savedOP.save()
+    else
+      savedOP = OrderProcessing.where(:order_id => orderPayload["order_id"]).first(1)[0]
+      if savedOP.publish_time <= orderPayload["publish_time"]
```

Comment:
why use this condition to verify existence?

Reply:
The exercise says that events may appear not in chronicle order therefore I want to only update the order if the publish_time is greater than the current publish_time